src/rc_allocator.c

```diff
@@ -236,6 +236,45 @@ rc_allocator_config_init(rc_allocator_config *allocator_cfg,
    allocator_cfg->extent_capacity = capacity / io_cfg->extent_size;
 }
 
+/*
+ *----------------------------------------------------------------------
+ * rc_allocator_valid_config() --
+ *
+ * Do minimal validation of RC-allocator cofiguration.
+ *----------------------------------------------------------------------
+ */
+platform_status
+rc_allocator_valid_config(rc_allocator_config *cfg)
+{
+   platform_status rc = STATUS_OK;
+   rc                 = laio_config_valid(cfg->io_cfg);
+   if (!SUCCESS(rc)) {
+      return rc;
+   }
+
+   if (cfg->capacity == 0) {
+      platform_error_log("Configured disk size %lu bytes is invalid.\n",
+                         cfg->capacity);
+      return STATUS_BAD_PARAM;
+   }
+   if (cfg->extent_capacity == 0) {
+      platform_error_log("Configured extent capacity %lu bytes is invalid.\n",
+                         cfg->extent_capacity);
+      return STATUS_BAD_PARAM;
+   }
+   if (cfg->capacity != (cfg->io_cfg->extent_size * cfg->extent_capacity)) {
```

Comment:
Not sure how I feel about this check.

This is guaranteed to be true of configs created by `rc_allocator_config_init`.

I guess this is just at init time, so it's ok.

But, if you're going to replicate constraints that are established by `config_init`, then might as well also check
- that page_size divides extent_size
- that page_capacity * page_size == capacity

Reply:
@rtjohnso - Here is what I did:

1. This one is currently not testable, as `laio_config_valid()` implements checks that we only support a fixed page-size and extent-size. So this division check is already enforced. [ I can't add any further checks as the can't be tested with the current code-flow. When we do allow different page-sizes / extent-sizes, such checks will need to be added. ]
2. Added this check, and added new test case `test_disk_size_not_integral_multiple_of_page_size` for this check.